packages/openid4vc/src/shared/issuerMetadataUtils.ts

```diff
@@ -41,43 +183,29 @@ export function getOfferedCredentials(
       )
     }
 
-    const foundSupportedCredential = allCredentialsSupported.find(
-      (supportedCredential): supportedCredential is OpenId4VciCredentialSupportedWithId =>
-        supportedCredential.id !== undefined && supportedCredential.id === offeredCredential
-    )
-
     // Make sure the issuer metadata includes the offered credential.
-    if (!foundSupportedCredential) {
+    const credentialConfigurationSupported = uniformCredentialsSupported[offeredCredential]
+    if (!credentialConfigurationSupported) {
       throw new Error(
         `Offered credential '${offeredCredential}' is not part of credentials_supported of the issuer metadata.`
       )
     }
-
-    credentialsSupported.push(foundSupportedCredential)
+    credentialConfigurationsOffered[offeredCredential] = credentialConfigurationSupported
   }
 
-  return credentialsSupported
-}
-
-// copied from sphereon as the method is only available on the client
-export function handleAuthorizationDetails(
-  authorizationDetails: AuthorizationDetails | AuthorizationDetails[],
-  metadata: EndpointMetadataResult
-): AuthorizationDetails | AuthorizationDetails[] | undefined {
-  if (Array.isArray(authorizationDetails)) {
-    return authorizationDetails.map((value) => handleLocations(value, metadata))
-  } else {
-    return handleLocations(authorizationDetails, metadata)
-  }
+  return credentialConfigurationsOffered
 }
 
-// copied from sphereon as the method is only available on the client
-function handleLocations(authorizationDetails: AuthorizationDetails, metadata: EndpointMetadataResult) {
-  if (typeof authorizationDetails === 'string') return authorizationDetails
-  if (metadata.credentialIssuerMetadata?.authorization_server || metadata.authorization_endpoint) {
-    if (!authorizationDetails.locations) authorizationDetails.locations = [metadata.issuer]
-    else if (Array.isArray(authorizationDetails.locations)) authorizationDetails.locations.push(metadata.issuer)
-    else authorizationDetails.locations = [authorizationDetails.locations as string, metadata.issuer]
+export const getProofTypesSupported = (agentContext: AgentContext) => {
+  return {
+    jwt: {
+      proof_signing_alg_values_supported: getSupportedJwaSignatureAlgorithms(agentContext) as string[],
+    },
+    cwt: {
+      proof_signing_alg_values_supported: [],
+    },
+    ldp_vp: {
+      proof_signing_alg_values_supported: [],
+    },
```

Comment:
Sphereon's types currently require it.

Reply:
Well then we need to make a PR and add a ts-ignore for now. As it would be incorrect to include them. including the key indicates you support the proof types.

Reply:
I mean not including any alg_values could indicate you don't support it, but still it doesn't make sense to include it